### PR TITLE
#754 Align And and Or ctors

### PR DESCRIPTION
**Pull request by pbenety**
Align And and Or ctors for #754

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#851](https://github.com/yegor256/cactoos/issues/851) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#851](https://github.com/yegor256/cactoos/issues/851) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/C63314D6Z/1300026004); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/java/org/cactoos/scalar/And.java [hunk 1]
```diff
@@ -116,7 +128,7 @@
      * @since 0.24
      */
     public <X> And(final Proc<X> proc, final Iterable<X> src) {
-        this(new FuncOf<>(proc, true), src);
+        this(new FuncOf<>(proc, false), src);
```
Comment (Vatavuk) [inline src/main/java/org/cactoos/scalar/And.java]: @pbenety this should be `this(new FuncOf<>(proc, true), src)`

File: src/test/java/org/cactoos/scalar/AndTest.java [hunk 2]
```diff
@@ -81,86 +79,115 @@ public void allFalse() throws Exception {
...
-                    Matchers.equalTo(true),
-                    new MatcherOf<>(
-                        value -> {
-                            return list.isEmpty();
-                        }
-                    )
-                )
-            )
+            list.size(),
+            Matchers.equalTo(1)
```
Comment (Vatavuk) [inline src/test/java/org/cactoos/scalar/AndTest.java]: @pbenety  Use `Matchers.contains()`

File: src/test/java/org/cactoos/scalar/AndTest.java [hunk 3]
```diff
@@ -81,86 +79,115 @@ public void allFalse() throws Exception {
...
-            new ScalarHasValue<>(
-                Matchers.allOf(
-                    Matchers.equalTo(true),
-                    new MatcherOf<>(
-                        value -> list.size() == 2
-                    )
-                )
-            )
+            list.size(),
+            Matchers.equalTo(1)
```
Comment (Vatavuk) [inline src/test/java/org/cactoos/scalar/AndTest.java]: @pbenety Size should be 2, this is a bug due to the wrong boolean parameter in the ctor that I've commented.
Use `Matchers.contains() `method instead for comparing entire array content.

File: src/test/java/org/cactoos/scalar/AndTest.java [hunk 4]
```diff
@@ -81,86 +79,115 @@ public void allFalse() throws Exception {
...
         final List<Integer> list = new LinkedList<>();
         new And(
             (Proc<Integer>) list::add,
-            1, 1
+            2, 3, 4
         ).value();
         MatcherAssert.assertThat(
             list.size(),
-            Matchers.equalTo(2)
+            Matchers.equalTo(3)
```
Comment (Vatavuk) [inline src/test/java/org/cactoos/scalar/AndTest.java]: @pbenety  Use `Matchers.contains()`

File: src/test/java/org/cactoos/scalar/OrTest.java [hunk 5]
```diff
@@ -111,6 +112,19 @@ public void testProc() throws Exception {
...
+    @Test
+    public void testProcIterator() throws Exception {
+        final List<Integer> list = new LinkedList<>();
+        new Or(
+            (Proc<Integer>) list::add,
+            new IteratorOf<>(1, 2, 3, 4)
+        ).value();
+        MatcherAssert.assertThat(
+            list.size(),
+            Matchers.equalTo(4)
```
Comment (Vatavuk) [inline src/test/java/org/cactoos/scalar/OrTest.java]: @pbenety  Use `Matchers.contains()`

**Comment by pbenety** [comment]
@Vatavuk done

**Comment by Vatavuk** [comment]
@llorllale good to merge

**Comment by llorllale** [comment]
@pbenety I note inconsistent specification for `@since` tags. Please set them all to `0.34` on all new ctors.

**Comment by 0crat** [comment]
@vatavuk[/z](https://www.0crat.com/u/vatavuk) this job was assigned to you 7days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by Vatavuk** [comment]
@0crat wait for merge

**Comment by 0crat** [comment]
> @0crat wait for merge ([here](https://github.com/yegor256/cactoos/issues/851#issuecomment-389523702))

@Vatavuk It's a code review job [#851](https://github.com/yegor256/cactoos/issues/851), can't put it on hold

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#851](https://github.com/yegor256/cactoos/issues/851) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@pbenety According to our Policy:
> All commit messages have links to issues

Please, confirm that you'll add a link to issue in all your future commits, including fixes. Thanks!

**Comment by elenavolokhova** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @vatavuk[/z](https://www.0crat.com/u/vatavuk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**